src/torrent.cpp

```diff
@@ -12253,7 +12269,7 @@ namespace libtorrent
 		TORRENT_ASSERT(is_single_thread());
 		INVARIANT_CHECK;
 
-		return (std::max)(0, int(m_connections.size()) - m_num_seeds - m_num_connecting);
```

Comment:
this needs to deduct the number of connecting *non-seed* peers as well:

return std::max(0, int(m_connections.size()) - m_num_seeds - m_num_connecting + m_num_connecting_seeds);